### PR TITLE
Refactor evaluation pipeline for gaussian-splatting

### PR DESCRIPTION
**Pull request by ginazhouhuiwu**
Changes:
- added `get_outputs_for_camera` to streamline eval pipeline for both ray-based and camera-based (e.g. gaussian) model
- changed `EvalDataloader` to output cameras instead of raybundles
- changed `next_eval_image` in all datamangers to output cameras instead of raybundles

TODO:
- track `metrics_dict["image_idx"]` during evaluation

### CODE REVIEW

**Comment by ginazhouhuiwu** [comment]
One more todo: currently metrics are not keeping track of `image_idx`. It's not affecting evaluation but there might need to be some changes in `Cameras` if we want to keep logging this.

**Comment by maturk** [comment]
Hey @ginazhouhuiwu, nice work. What is the status of this pr atm?

**Comment by ginazhouhuiwu** [comment]
@maturk Waiting for a review. Meanwhile if you want to use gsplat evaluation logging this branch works fine!

File: nerfstudio/pipelines/base_pipeline.py [hunk 1]
```diff
@@ -383,21 +375,22 @@ def get_average_eval_image_metrics(
...
                 metrics_dict, images_dict = self.model.get_image_metrics_and_images(outputs, batch)
 
-                if output_path is not None:
-                    camera_indices = camera_ray_bundle.camera_indices
-                    assert camera_indices is not None
-                    for key, val in images_dict.items():
-                        Image.fromarray((val * 255).byte().cpu().numpy()).save(
-                            output_path / "{0:06d}-{1}.jpg".format(int(camera_indices[0, 0, 0]), key)
-                        )
+                # if output_path is not None:
```
Comment (maturk) [inline nerfstudio/pipelines/base_pipeline.py:386]: Also here, should these be commented?

File: nerfstudio/viewer_beta/render_state_machine.py [hunk 2]
```diff
@@ -156,30 +155,23 @@ def _render_img(self, camera_state: CameraState):
...
+                            outputs = self.viewer.get_model().get_outputs_for_camera(camera=camera)
                 except viewer_utils.IOChangeException:
                     self.viewer.get_model().train()
                     raise
                 self.viewer.get_model().train()
-            if isinstance(self.viewer.get_model(), GaussianSplattingModel):
-                num_rays = image_height * image_width
-            else:
-                num_rays = len(camera_ray_bundle)
+            num_rays = (camera.height * camera.width * camera.size).item()
```
Comment (maturk) [inline nerfstudio/viewer_beta/render_state_machine.py:166]: Is this correct? just wondering, I dont understand how this gives the same as len(camera_ray_bundle) in raybundle models.

File: nerfstudio/configs/method_configs.py [hunk 3]
```diff
@@ -600,13 +600,13 @@
 
 method_configs["gaussian-splatting"] = TrainerConfig(
     method_name="gaussian-splatting",
-    steps_per_eval_image=100,
-    steps_per_eval_batch=100,
+    steps_per_eval_image=50,
+    steps_per_eval_batch=50,
     steps_per_save=2000,
-    steps_per_eval_all_images=100000, 
+    steps_per_eval_all_images=100,
```
[the remark this answers is not in the document]
Comment (ginazhouhuiwu) [reply nerfstudio/configs/method_configs.py:606]: oh yes! Forgot to change it back after debugging

File: nerfstudio/pipelines/base_pipeline.py [hunk 4]
```diff
@@ -341,19 +339,13 @@ def get_eval_image_metrics_and_images(self, step: int):
...
         self.eval()
-        image_idx, camera_ray_bundle, batch = self.datamanager.next_eval_image(step)
-        # TODO ginawu: refactor(?) for gsplat without this hardcoded camera param below
-        outputs = self.model.get_outputs_for_camera_ray_bundle(None, camera=camera_ray_bundle)
+        camera, batch = self.datamanager.next_eval_image(step)
+        outputs = self.model.get_outputs_for_camera(camera)
         metrics_dict, images_dict = self.model.get_image_metrics_and_images(outputs, batch)
-        assert "image_idx" not in metrics_dict
-        metrics_dict["image_idx"] = image_idx
+        # assert "image_idx" not in metrics_dict
```
[the remark this answers is not in the document]
Comment (ginazhouhuiwu) [reply nerfstudio/pipelines/base_pipeline.py:345]: Currently image indices are not tracked. Another refactor is needed here where the cameras store the indices

File: nerfstudio/pipelines/base_pipeline.py (hunk 1 above)
Comment (ginazhouhuiwu) [reply nerfstudio/pipelines/base_pipeline.py:386]: Same as above for image indices not being tracked at the moment.

File: nerfstudio/viewer_beta/render_state_machine.py (hunk 2 above)
Comment (ginazhouhuiwu) [reply nerfstudio/viewer_beta/render_state_machine.py:166]: I believe I took this from elsewhere in the codebase but I'm not completely sure if it's accurate.

**Comment by pwais** [comment]
just ran into a crash that this PR fixes. thank you so much for doing this PR !!! 🎉

**Pull request merged**